apps/createsamples/utility.cpp

```diff
@@ -1270,7 +1270,7 @@ void cvCreateTrainingSamples( const char* filename,
                 if( showsamples )
                 {
                     cvShowImage( "Sample", &sample );
-                    if( cvWaitKey( 0 ) == 27 )
+                    if( (cvWaitKey( 0 ) & 0xFF) == 27 )
```

Comment:
Just checked, it seems that on Ubuntu calling the code 

```
using namespace std;

int main( int argc, const char** argv )
{
    int key = 0;
    key = 'ESC';
    cerr << key << endl;

    return 0;
}
```

yields the same output as you!